### DISCUSSION TITLE
Function to convert IP Address w.x.y.z to integer?

### DISCUSSION THREAD

**Discussion by richb-hanover** [pawelsalawa/sqlitestudio#4063] [Q&A] [answered]
I'd like to import a file of IPv4 addresses in the form w.x.y.z (e.g. 192.168.1.1) and convert them to the corresponding value as an integer.

Does anyone have a function that can do this? Many thanks.

PS I have seen the reverse function (integer -> string) many places, for example: https://sqlite-users.sqlite.narkive.com/mC4pSHS7/convert-integer-ip-address-to-string-ip-address-in-sqlite

**Comment by pawelsalawa** [comment, marked as answer]
You can go to SQLiteStudio's custom functions (https://github.com/pawelsalawa/sqlitestudio/wiki/User_Manual#custom-sql-functions) and add these functions:

https://github.com/pawelsalawa/sqlitestudio/wiki/Qt_script_ip2int
https://github.com/pawelsalawa/sqlitestudio/wiki/Qt_script_int2ip

Remember that such functions work only when you operate on the database from the SQLiteStudio. If you want such functions in your own application using SQLite, you need to register & implement them in your application.

**Reply by richb-hanover** [reply]
This is great! It definitely makes me hit the [Donate](https://sqlitestudio.pl/donate/) button! :-) Thanks.

**Comment by richb-hanover** [comment]
A follow-on question: How can I find other functions that have been created? I'm teaching myself SQL, and this is a fun tool!

**Reply by pawelsalawa** [reply]
I understand you're asking for a SQLite built-in functions. These are listed & described at https://sqlite.org/lang_corefunc.html and its subpages.

**Reply by richb-hanover** [reply]
I know about the SQLite built-in functions. I was looking for a list of the other goodies that you or others have created, similar to ip2int and int2ip... Thanks.

**Reply by pawelsalawa** [reply]
I see. There is not a lot of it, but there is a little bit:
https://github.com/pawelsalawa/sqlitestudio/wiki/Scripts_repository

**Comment by richb-hanover** [comment]
Perfect! I wonder how I could have found that pagemyself? Perhaps the SQLiteStudio's custom functions page (https://github.com/pawelsalawa/sqlitestudio/wiki/User_Manual#custom-sql-functions) could have a link? Thanks again

**Reply by pawelsalawa** [reply]
It's available directly from the main wiki page: https://github.com/pawelsalawa/sqlitestudio/wiki 
![image](https://user-images.githubusercontent.com/481846/114705670-4366df00-9d28-11eb-897d-08ad8a7db8f7.png)

Still, it's a good idea to add it where you suggested and I will do it in a moment.

**Reply by richb-hanover** [reply]
Thanks, again!
